### ISSUE TITLE
Stack.reset_parameters() undoes structure a sub-layer's constructor established

### ISSUE THREAD

**Issue by KennethGrace** [latticedynamics/lumen#8]
`Stack.reset_parameters()` re-draws **every** `nn.Linear.weight` in the trunk.
Any sub-layer whose constructor established *structure among* its linear weights
therefore loses it the moment that sub-layer is placed in a `Stack`, silently —
the model trains, and simply does not start where its own documentation says it
starts.

`decay="state_gated"` is the instance that exists today.

## Reproduction

```python
import torch
from lumen import GatedDeltaNet, GatedDeltaNetConfig, HeadLayout
from lumen.block import Block
from lumen.stack import Stack

config = GatedDeltaNetConfig(d_model=64, layout=HeadLayout.shared_key(4),
                             expand_k=1.0, decay="state_gated")

torch.manual_seed(0)
standalone = GatedDeltaNet(config)
rows = standalone.a_proj.weight
print("standalone      :", bool(torch.all(rows == rows[0])))

torch.manual_seed(0)
trunk = Stack(64, 1, lambda i: Block(64, GatedDeltaNet(config),
                                     norm_eps=1e-5, d_mlp=0), norm_eps=1e-5)
rows = trunk.blocks[0].mixer.a_proj.weight
print("inside a Stack  :", bool(torch.all(rows == rows[0])),
      f"  max|row0-row1| = {(rows[0] - rows[1]).abs().max():.4f}")
```

```
standalone      : True
inside a Stack  : False   max|row0-row1| = 0.0743
```

## What is lost

`GatedDeltaNet.__init__` states the guarantee and implements it:

> Every state in a key group starts on the **same** gate, so the layer begins in
> the `key_group` arrangement and has to earn its way out of it

```python
shared = self.a_proj.weight[:n_key_groups].clone()
self.a_proj.weight.copy_(shared.repeat_interleave(m, dim=0))
```

`Stack.reset_parameters()` then runs `nn.init.normal_(module.weight,
std=self.init_std)` over `self.modules()`, which includes `a_proj`, and the
arrangement is gone.

## Why it is easy to miss

The failure is **asymmetric across two arrangements that are otherwise
siblings**, and it lands on the weaker guarantee:

| | mechanism | survives a `Stack`? |
|---|---|---|
| `decay="state"` | zero-init `a_offset`, a bare `nn.Parameter` | **yes** |
| `decay="state_gated"` | structured `a_proj.weight`, an `nn.Linear` | **no** |

Deliberate biases are unaffected, exactly as `reset_parameters()` documents —
`a_proj.bias` is still −3.0 — so the layer looks correctly initialised under the
obvious spot-check.

And `tests/test_gdn.py::test_state_gated_starts_in_the_key_group_arrangement`
passes, because it builds the layer standalone. The guarantee is tested in the
only configuration where it holds.

## This is the sibling of a hazard already reasoned about

`Stack.reset_parameters()`'s docstring works through precedence for the **depth**
pass:

> The depth pass runs second and therefore wins, including over a sub-layer that
> deliberately initialised its own output projection — `UndertowConfig(zero_init=True)`
> being the case that exists. That is the documented precedence rather than an
> accident

That reasoning is right, and it has a test
(`tests/test_undertow.py::test_zero_init_collides_with_depth_scaled_init`). The
gap is that the **base** pass has the same hazard and is broader — it touches
every `nn.Linear` in the trunk, not only the residual writes — and it is not
addressed anywhere.

So this is arguably not a new problem but the unhandled half of a known one.

## Note on severity

For a gradient-trained consumer this is a mild init difference. For a consumer
that does **not** train by gradient — anything population-based — the initial
weights *are* the prior the entire search is drawn from, and there is no
gradient to walk a wrong one back. A silently altered init there does not shift
a starting point; it shifts the distribution being searched, and makes an arm
non-comparable to its siblings with nothing to flag it.

That is a reason to prefer a fix that generalises over one that patches the
symptom, not a reason to hurry.

## Possible shapes, not a recommendation

1. **A `self_initialised_parameters()` contract** that `Stack` skips, in the same
   refuse-to-guess style as `residual_out_projections()`. The only shape that
   also covers the next sub-layer to want this.
2. **Call sub-layer `reset_parameters()` after the base pass**, making the
   ordering explicit. Requires every sub-layer to grow one.
3. **Document the precedence and have the mixer re-apply**, matching how the
   `zero_init` collision was resolved — cheapest, and leaves the general case
   open.

Whichever lands, the test needs to construct the layer **inside a `Stack`**;
the standalone assertion cannot observe this.

## Open question for the record

Is the base pass initialising sub-layer internals *at all* the right default?
`Stack` exists to own the depth-scaled rescale, which genuinely cannot live
lower down. Re-drawing every weight in the trunk is a separate policy that came
along with it, and a sub-layer arguably knows better than the stack does how its
own projections should start.
